Runtime/AvatarCreator/UI/Elements/AssetSelectionElement.cs

```diff
@@ -0,0 +1,62 @@
+using System.Linq;
+using System.Threading.Tasks;
+using ReadyPlayerMe.Core;
+using UnityEngine;
+
+namespace ReadyPlayerMe.AvatarCreator
+{
+
+    /// <summary>
+    /// Manages the selection of assets for a given body type, gender, and asset type.
+    /// This class is responsible for loading asset data and creating corresponding buttons for each asset.
+    /// It extends the functionality of SelectionElement to handle PartnerAsset-specific interactions.
+    /// </summary>
+    public class AssetSelectionElement : SelectionElement
+    {
+        [Header("Properties")]
+        [SerializeField] private BodyType bodyType = BodyType.FullBody;
+        [SerializeField] private AssetType assetType;
+        [SerializeField] private int iconSize = 64;
+
+        private PartnerAsset[] partnerAssets;
```

Comment:
Could we just call this 'asset'? PartnerAsset implies it's some sort of join object between a Partner (or application) and an Asset. Actually this object has no property that references the partner so the naming doesn't make much sense IMO.

Reply:
updated